sharding-core/sharding-core-common/src/test/java/org/apache/shardingsphere/core/strategy/keygen/SnowflakeShardingKeyGeneratorTest.java

```diff
@@ -76,6 +76,21 @@ public void assertGenerateKeyWithSingleThread() {
         assertThat(actual, is(expected));
     }
     
+    @Test
+    public void assertGenerateKeyWithMaxVibrationOffsetBe1LessThanPowerOf2() {
+        SnowflakeShardingKeyGenerator keyGenerator = new SnowflakeShardingKeyGenerator();
+        Properties properties = new Properties();
+        properties.setProperty("max.vibration.offset", String.valueOf(3));
+        keyGenerator.setProperties(properties);
+        SnowflakeShardingKeyGenerator.setTimeService(new FixedTimeService(1));
+        List<Comparable<?>> expected = Arrays.<Comparable<?>>asList(1L, 4194306L, 4194307L, 8388611L, 8388612L, 12582912L, 12582913L, 16777217L, 16777218L, 20971522L);
+        List<Comparable<?>> actual = new ArrayList<>();
+        for (int i = 0; i < DEFAULT_KEY_AMOUNT; i++) {
+            actual.add(keyGenerator.generateKey());
+        }
+        assertThat(actual, is(expected));
+    }
+    
```

Comment:
The test case is still not clear, we just case about the last digital of generated keys, right?
The first case, we can assert the last digital of generated keys for each milliseconds as `0`, `1`, `2`, `3` and `0` again.
The second case, we can assert the last digital of generated keys in same milliseconds as accumulate over the Vibration offset.